httpclient-backend/fs2/src/main/scala/sttp/client/httpclient/fs2/HttpClientFs2Backend.scala

```diff
@@ -0,0 +1,94 @@
+package sttp.client.httpclient.fs2
+
+import java.io.InputStream
+import java.net.http.{HttpClient, HttpRequest}
+import java.net.http.HttpRequest.BodyPublishers
+import java.nio.ByteBuffer
+
+import cats.effect._
+import cats.effect.implicits._
+import cats.implicits._
+import fs2.Stream
+import fs2.interop.reactivestreams._
+import org.reactivestreams.FlowAdapters
+import sttp.client.{FollowRedirectsBackend, SttpBackend, SttpBackendOptions}
+import sttp.client.httpclient.{HttpClientAsyncBackend, HttpClientBackend, WebSocketHandler}
+import sttp.client.impl.cats.implicits._
+import sttp.client.testing.SttpBackendStub
+
+import scala.util.{Success, Try}
+import sttp.client.ws.WebSocketResponse
+
+class HttpClientFs2Backend[F[_]: ConcurrentEffect: ContextShift] private (
+    client: HttpClient,
+    blocker: Blocker,
+    chunkSize: Int,
+    closeClient: Boolean,
+    customizeRequest: HttpRequest => HttpRequest
+) extends HttpClientAsyncBackend[F, Stream[F, ByteBuffer]](client, implicitly, closeClient, customizeRequest) {
+
+  override def openWebsocket[T, WS_RESULT](
+      request: sttp.client.Request[T, Stream[F, ByteBuffer]],
+      handler: WebSocketHandler[WS_RESULT]
+  ): F[WebSocketResponse[WS_RESULT]] =
+    super.openWebsocket(request, handler).guarantee(ContextShift[F].shift)
+
+  override def streamToRequestBody(stream: Stream[F, ByteBuffer]): HttpRequest.BodyPublisher =
+    BodyPublishers.fromPublisher(FlowAdapters.toFlowPublisher(stream.toUnicastPublisher()))
+
+  override def responseBodyToStream(responseBody: InputStream): Try[Stream[F, ByteBuffer]] =
+    Success(fs2.io.readInputStream(responseBody.pure[F], chunkSize, blocker).chunks.map(_.toByteBuffer))
+}
+
+object HttpClientFs2Backend {
+  private val defaultChunkSize: Int = 4096
+
+  private def apply[F[_]: ConcurrentEffect: ContextShift](
+      client: HttpClient,
+      blocker: Blocker,
+      chunkSize: Int,
+      closeClient: Boolean,
+      customizeRequest: HttpRequest => HttpRequest
+  ): SttpBackend[F, Stream[F, ByteBuffer], WebSocketHandler] =
+    new FollowRedirectsBackend(new HttpClientFs2Backend(client, blocker, chunkSize, closeClient, customizeRequest))
+
+  def apply[F[_]: ConcurrentEffect: ContextShift](
+      blocker: Blocker,
```

Comment:
in the http4s backend, we use a default value for blocker of `Blocker.liftExecutionContext(ExecutionContext.Implicits.global)`.

in the httpclient-monix one, similarly we use a defualt scheduler: `Scheduler.Implicits.global`. 

Maybe here we could add the default as well?

Reply:
As far as I understand, using the global execution context for blocking is not a good idea.

See the comment here:
https://github.com/typelevel/cats-effect/blob/81bff417cade6c0ccfa6d3a1c32be32381a9ed70/core/shared/src/main/scala/cats/effect/Blocker.scala#L57-L63

Creating a cached thread pool with `Blocker.apply` in one place in your application, and passing that down, is the preferred method, iiuc.

Reply:
I suppose you're right, I'll remove blocker from http4s as well :)